WalletWasabi/WabiSabi/Client/CoinJoinTrackingDataFactory.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections.Generic;
+using System.Threading;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Wallets;
+using WalletWasabi.WebClients.Wasabi;
+
+namespace WalletWasabi.WabiSabi.Client;
+
+public class CoinJoinTrackingDataFactory
+{
+	public CoinJoinTrackingDataFactory(
+		IWasabiHttpClientFactory httpClientFactory,
+		RoundStateUpdater roundStatusUpdater,
+		CancellationToken cancellationToken)
+	{
+		HttpClientFactory = httpClientFactory;
+		RoundStatusUpdater = roundStatusUpdater;
+		CancellationToken = cancellationToken;
```

Comment:
If cancellation token is removed from ctor and put in `CreateCoinJoinTrackingData` method, then it looks like you can have just one instance of `CoinJoinTrackingDataFactory` in `CoinJoinManager` and it would feel like a more natural factory.

Reply:
This is already done, I have only one!